src/electrum/server.rs

```diff
@@ -490,6 +501,17 @@ impl Connection {
         Ok(result)
     }
 
+    fn log_rpc_event(&self, mut log: Value) {
+        log.as_object_mut().unwrap().insert(
+            "source".into(),
+            json!({
+                "ip": self.addr.ip().to_string(),
+                "port": self.addr.port(),
+            }),
+        );
+        println!("ELECTRUM-RPC-LOGGER: {}", json!(log));
```

Comment:
The `json!()` here isn't necessary since `log` already is a json Value

All LGTM other than that. thanks!

Reply:
Right, corrected.